### DISCUSSION TITLE
Subtitles and Accessibility Features?

### DISCUSSION THREAD

**Discussion by Branpanman** [ihhub/fheroes2#4172] [Ideas]
Hello! I’m Brandon. I’m a long time HoMM fan and seeing HoMM 2 come to new life in this project is truly the Lord’s Work… I’m so excited about it that I even signed up to GitHub just for it!

I’m interested in contributing to this project, either via Patreon or through lending technical assistance (10+ years working IT - though not an engineer - and my coding skills are serviceable). 

The area I’d like most to contribute to - and my question/idea for this thread - is actually about accessibility: have any features been considered yet to make the game more accessible to people with special needs? 

Myself, I am hard of hearing, and I’ve always struggled to understand the spoken dialogue in the videos in HoMM 2 because they don’t have any subtitles/captions. I typically rely on captions/subtitles to fully enjoy a gaming experience, and the lack of these us held me back with HoMM 2&3. 

So has anyone on the dev team thought of adding these features? Other “low-bar” accessibility features include colorblind-friendly palette, and difficulty settings (the latter of which are already included in the game). 

Anyway, really excited to see this project progress and to help contribute in any way that I can. 

Looking forward to connecting with folks here.

**Comment by ihhub** [comment]
Hi @Branpanman  and welcome to the project!

There are many areas to contribute to the project:
- **development**. The most wide area. Basic programming skills are required (even if it's not C++ language) to solve some issues. We have a huge variety of bugs to be fixed and features to be implemented. You can try your skills by picking issues which have **Good first issue** label on them. The project has automated CI/CD pipeline which helps any developer to find some weak places in their code.
- **testing**. Since we have a lot of changes being committed we struggle in time to test all of them. First of all, not all bugs or missing features have been discovered. Secondly, each pull request from a developer ideally should go through testing stage before being merged. It could look like an easy area but it's very tedious as it requires a lot of patience.
- **design**. We're going to expand the game by introducing a brand new add-on. Our team is lacking of very good pixel art designers who can help us adding new content. So a person who is willing to help with pixel art would receive a bunch of tasks to work with starting from a simple one image design.
- **sound and music**. The original game is missing sounds for certain objects on the map. Also we require new sound and music content to be added in the aforementioned add-on. Nothing more to say about this area as we're still on a trial stage here.
- **translation**. The main language in the game is English but our player base come from many countries. We have support of Polish, Russian, French and now German languages which still require help to finish all translations. In the near future we're going to expand the list of supported languages to cover ever wider audience.
- **support**. Anyone can support the project either by spreading the word about it or making a donation (via Patreon) which will used for the future improvements of the project.

We made few quality of life changes to make the game more playable and in relation to readability we added much more text information for multiple places in the game. For example, extra popup windows for UI elements (when you right click on them) or instant battles.

Using a different palette within the game is totally doable. We just need to know which palette to use. I personally see no issue in adding features for people with special needs, we just need to come up with something doable and helpful in this case.

**Reply by ihhub** [reply]
Edited the message above as I missed to write about translations :)

**Reply by Branpanman** [reply]
This is fantastic! Thank you @ihhub I am looking forward to contributing to this project however I can. I'll continue the discussion with you on the Discord :)

**Comment by LeHerosInconnu** [comment]
Welcome @Branpanman and hello @ihhub,

The texts of all the original campaigns of the game are present in the two files "H2CAMP.TXT" and "POLCAMP.TXT". :)
These two files are included with the English Gold version and the GoG version of the game.
